### ISSUE TITLE
Building .NET 7 android binding gives error APT2144: invalid file path '(...)/obj/Debug/net7.0-android/lp/149.stamp'

### ISSUE THREAD

**Issue by F-Goncalves** [xamarin/xamarin-android#7660]
### Android application type

Android for .NET (net6.0-android, etc.)

### Affected platform version

VS 2022 17.4.2, .NET 7.0.101, Xamarin Android 13.1.0.1

### Description

Hi,

I am trying to migrate a Xamarin Android binding to .NET 7 and am seeing following errors during build time:

```
/usr/local/share/dotnet/packs/Microsoft.Android.Sdk.Darwin/33.0.4/tools/Xamarin.Android.Aapt2.targets(123,3): error APT2144: invalid file path '(...)/obj/Debug/net7.0-android/lp/149.stamp
/usr/local/share/dotnet/packs/Microsoft.Android.Sdk.Darwin/33.0.4/tools/Xamarin.Android.Aapt2.targets(123,3): error APT2144: invalid file path '(...)/obj/Debug/net7.0-android/lp/158.stamp'
```

But the files that cannot be found do exist.

Any help on this issue would be appreciated.

### Steps to Reproduce

-

**Comment by F-Goncalves** [comment]
Unfortunately it makes no difference. Honestly I do not even have a clue what the .stamp files are used for.

**Comment by jpobst** [comment]
@dellis1972 Seems like these `.stamp` files should not be passed to `AAPT`?

**Comment by F-Goncalves** [comment]
@dellis1972 @jpobst Any news on this?

**Comment by dellis1972** [comment]
@F-Goncalves can you share the Project? (or just even the csproj) It would help us repo the issue.

**Comment by F-Goncalves** [comment]
@dellis1972 
I just invited you to a private repository. Since it is using SlimBinding approach from https://github.com/Redth/Xamarin.Binding.Helpers you will have to build the Android Studio project first, before you can build the binding.

**Comment by dellis1972** [comment]
@F-Goncalves 

to it looks like we have duplicates of `react-native-0.68.5.aar` being used somehow. 
I tried changing all the `LibraryProjectZip` entires to the new `AndroidLibrary` and setting the `Bind` attribute to true

```
<ItemGroup>
        <AndroidLibrary Include="Jars\react-native-0.68.5.aar" />
    </ItemGroup>
```


but that resulted in compilation errors. 

```
Com.Facebook.React.Maui/obj/Debug/net8.0-android/generated/src/Com.Facebook.React.ReactActivity.cs(20,170): error CS0738: 'ReactActivity' does not implement interface member 'IPermissionAwareActivity.CheckPermission(string?, int, int)'. 'ContextWrapper.CheckPermission(string?, int, int)' cannot implement 'IPermissionAwareActivity.CheckPermission(string?, int, int)' because it does not have the matching return type of 'int'.
```

Are you wanting to bind the `react-native-0.68.5.aar` or just consume it in the final `Slim` binding?

**Comment by F-Goncalves** [comment]
@dellis1972 
I want to consume it in the final Slim Binding. Did you also encounter the error with the .stamp files before you made theses changes?

**Comment by dellis1972** [comment]
@F-Goncalves I did, it seems to be the duplicate `react-native-0.68.5.aar` which are causing the issue. I'm trying to figure out how to fix it. 

Do you actually bind need to the C# code for `react-native-0.68.5.aar`? or are you litterally just consuming the `react-native-0.68.5.aar` without binding it.

**Comment by F-Goncalves** [comment]
@dellis1972 
I don't call into the `react-native-0.68.5.aar`. The JitsiMeetSdk I am trying to bind depends on it. If there's a way I can omit the additional BindingProject for React, I would be happy to do that.

Thanks for taking the time to help!

**Comment by dellis1972** [comment]
@F-Goncalves 

I just did a test. 

I completely removed the reference to `Org.Jitsi.Meet.Sdk.Maui.csproj` from `JitsiMeetSdk.SlimBinding.Android.Maui.csproj`. And it built fine. I don't think you need the other two binding projects at all. You just need to SlimBinding one. The BindingHelper thing @Redth wrote will pick up the dependencies for you.

**Comment by dellis1972** [comment]
the only reason to have the other two projects is if you are using the api. I suspect though that the reason you want to use the slim binding is so you can limit the C# code generated to just the bits you want. i.e the API you surface in `mylibrary-release.aar`. So unless you actually use the react API in C# I suspect you can remove those two completely and let the BindingHelper thing do the hard work for you.

**Comment by F-Goncalves** [comment]
@dellis1972 
That sounds good. I need to see if it also does what it's supposed to when running. I still have a feeling there was a reason I had to add the additional projects.

Were you also able to build the Android app project?
Lately I am getting a NullReferenceException in `BINDINGSGENERATOR`

```
BINDINGSGENERATOR : error BG0000: System.NullReferenceException: Object reference not set to an instance of an object.
       at MonoDroid.Generation.MethodBase.<>c.<AutoDetectEnumifiedOverrideParameters>b__63_0(Parameter p) in /Users/runner/work/1/s/xamarin-android/external/Java.Interop/tools/generator/Java.Interop.Tools.Generator.ObjectModel/MethodBase.cs:line 37
       at System.Linq.Enumerable.All[TSource](IEnumerable`1 source, Func`2 predicate)
       at MonoDroid.Generation.MethodBase.AutoDetectEnumifiedOverrideParameters(AncestorDescendantCache cache) in /Users/runner/work/1/s/xamarin-android/external/Java.Interop/tools/generator/Java.Interop.Tools.Generator.ObjectModel/MethodBase.cs:line 66
       at MonoDroid.Generation.GenBase.UpdateEnums(CodeGenerationOptions opt, AncestorDescendantCache cache) in /Users/runner/work/1/s/xamarin-android/external/Java.Interop/tools/generator/Java.Interop.Tools.Generator.ObjectModel/GenBase.cs:line 859
       at MonoDroid.Generation.GenBase.UpdateEnums(CodeGenerationOptions opt, AncestorDescendantCache cache) in /Users/runner/work/1/s/xamarin-android/external/Java.Interop/tools/generator/Java.Interop.Tools.Generator.ObjectModel/GenBase.cs:line 859
       at MonoDroid.Generation.GenBase.UpdateEnums(CodeGenerationOptions opt, AncestorDescendantCache cache) in /Users/runner/work/1/s/xamarin-android/external/Java.Interop/tools/generator/Java.Interop.Tools.Generator.ObjectModel/GenBase.cs:line 859
       at MonoDroid.Generation.GenBase.UpdateEnums(CodeGenerationOptions opt, AncestorDescendantCache cache) in /Users/runner/work/1/s/xamarin-android/external/Java.Interop/tools/generator/Java.Interop.Tools.Generator.ObjectModel/GenBase.cs:line 859
       at Xamarin.Android.Binder.CodeGenerator.Run(CodeGeneratorOptions options, DirectoryAssemblyResolver resolver) in /Users/runner/work/1/s/xamarin-android/external/Java.Interop/tools/generator/CodeGenerator.cs:line 257
       at Xamarin.Android.Binder.CodeGenerator.Run(CodeGeneratorOptions options) in /Users/runner/work/1/s/xamarin-android/external/Java.Interop/tools/generator/CodeGenerator.cs:line 50
       at Xamarin.Android.Binder.CodeGenerator.Main(String[] args) in /Users/runner/work/1/s/xamarin-android/external/Java.Interop/tools/generator/CodeGenerator.cs:line 33
```

**Comment by dellis1972** [comment]
@F-Goncalves the android sample was not in the repo. I'll do a pull and see if its been added

**Comment by F-Goncalves** [comment]
@dellis1972 
Oh I see. I will push it.

**Comment by F-Goncalves** [comment]
Done

**Comment by F-Goncalves** [comment]
Hey @dellis1972,
it's been a while. Have you been able to compile the sample app?

**Comment by dellis1972** [comment]
@F-Goncalves I am seeing the same `BINDINGSGENERATOR : error BG0000: System.NullReferenceException: Object reference not set to an instance of an object.` error locally. 

We might need to get help from @jpobst on that one, so you might need to add him to the private repo as well.

**Comment by F-Goncalves** [comment]
@jpobst I invited you to the repo. Would be awesome if you could help here. Thanks in advance

**Comment by jpobst** [comment]
I still get the original errors, was there a fix committed for them?

```
C:\Program Files\dotnet\packs\Microsoft.Android.Sdk.Windows\33.0.26\tools\Xamarin.Android.Aapt2.targets(123,3): error A
PT2144: invalid file path 'C:\code\temp\JitsiMeetSdk.Maui\JitsiMeetSdk.SlimBinding.Android.Maui\obj\Debug\net7.0-androi
d\lp\158.stamp'. [C:\code\temp\JitsiMeetSdk.Maui\JitsiMeetSdk.SlimBinding.Android.Maui\JitsiMeetSdk.SlimBinding.Android
.Maui.csproj]
C:\Program Files\dotnet\packs\Microsoft.Android.Sdk.Windows\33.0.26\tools\Xamarin.Android.Aapt2.targets(123,3): error A
PT2144: invalid file path 'C:\code\temp\JitsiMeetSdk.Maui\JitsiMeetSdk.SlimBinding.Android.Maui\obj\Debug\net7.0-androi
d\lp\158.stamp'. [C:\code\temp\JitsiMeetSdk.Maui\JitsiMeetSdk.SlimBinding.Android.Maui\JitsiMeetSdk.SlimBinding.Android
.Maui.csproj]
C:\Program Files\dotnet\packs\Microsoft.Android.Sdk.Windows\33.0.26\tools\Xamarin.Android.Aapt2.targets(123,3): error A
PT2144: invalid file path 'C:\code\temp\JitsiMeetSdk.Maui\JitsiMeetSdk.SlimBinding.Android.Maui\obj\Debug\net7.0-androi
d\lp\149.stamp'. [C:\code\temp\JitsiMeetSdk.Maui\JitsiMeetSdk.SlimBinding.Android.Maui\JitsiMeetSdk.SlimBinding.Android
.Maui.csproj]
```

**Comment by jpobst** [comment]
I think the NRE is the issue I found last week trying to bind the latest Google Material:
https://github.com/xamarin/java.interop/pull/1080

Due to the default includes in .NET SDK Style projects, you are trying to bind all 158 libraries in `\JitsiLibs`, which includes Google Material.

You can disable this with `<EnableDefaultItems>false</EnableDefaultItems>`.

Then you're down to only trying to incorrectly bind ~100 libraries in `obj\Debug\net7.0-android\library_project_jars`.  I guess these come from the slim bindings something?

In short, I would highly recommend not trying to use slim bindings infrastructure.  

I would suggest:
- Create a binding project and add `mylibrary-release.aar` as an `<AndroidLibrary>`.
- Add the Java dependencies that your library needs but shouldn't produce C# bindings as `<AndroidLibrary Bind='false'>`.

**Comment by andersonvieiragomeslopes** [comment]
Hello,
I hope I'm not late in the discussion, we are facing the same error with library bindings.

BINDINGSGENERATOR : error BG0000: System.NullReferenceException: Object reference not set to an instance of an object.

By insisting, we sometimes manage to generate the nuget package, but it breaks when trying to compile in a project, our project is public and we are using the azure communication services voip library.

![image](https://github.com/xamarin/xamarin-android/assets/26186056/1e8b54b1-f83a-4a5b-aafb-56f7c35d80bb)


https://github.com/Laerdal/Xamarin.AzureCommunicationCalling/tree/maui-beta/Bindings/Android/Laerdal.Maui.AzureCommunicationCalling.Android


@F-Goncalves were you able to find a solution to your problem?

**Comment by jpobst** [comment]
> BINDINGSGENERATOR : error BG0000: System.NullReferenceException: Object reference not set to an instance of an object.

This is a generic error so there's a decent chance it isn't related to this issue.

I would suggest filing a new issue for this, ideally with:
- Full stack trace of the exception
- Sample product that produces this error

As for the runtime error, that should likely be a separate issue.  Look at the output log to get the full text of the Java error.  (You may need to bump up the logging verbosity to see it, not sure.)

**Comment by andersonvieiragomeslopes** [comment]
> > BINDINGSGENERATOR : error BG0000: System.NullReferenceException: Object reference not set to an instance of an object.
> 
> This is a generic error so there's a decent chance it isn't related to this issue.
> 
> I would suggest filing a new issue for this, ideally with:
> 
> * Full stack trace of the exception
> * Sample product that produces this error
> 
> As for the runtime error, that should likely be a separate issue. Look at the output log to get the full text of the Java error. (You may need to bump up the logging verbosity to see it, not sure.)


It is an azure communication services library [binding project](https://github.com/Laerdal/Xamarin.AzureCommunicationCalling/tree/maui-beta/Bindings/Android/Laerdal.Maui.AzureCommunicationCalling.Android), we are migrating to maui, but we came across the error below, this generic error that I mentioned was overcome, but I ran into another one later, to reproduce just add the package nuget that I generate in the link project in any example and build.

1>Exception in thread "main" java.lang.RuntimeException: com.android.tools.r8.CompilationFailedException: Compilation failed to complete, origin: obj\Debug\net6.0-android33.0\lp\189\jl\libs\ED64959F88B22E6D.jar:com/google/gson/FieldNamingPolicy$5.class
1>	at com.android.tools.r8.internal.Bj.a(R8_3.3.28_2aaf796388b4e9f6bed752d926eca110512a53a3f09a8d755196089c1cfdf799:98)
1>	at com.android.tools.r8.D8.main(R8_3.3.28_2aaf796388b4e9f6bed752d926eca110512a53a3f09a8d755196089c1cfdf799:4)
1>Caused by: com.android.tools.r8.CompilationFailedException: Compilation failed to complete, origin: obj\Debug\net6.0-android33.0\lp\189\jl\libs\ED64959F88B22E6D.jar:com/google/gson/FieldNamingPolicy$5.class
1>	at Version.fakeStackEntry(Version_3.3.28.java:0)
1>	at com.android.tools.r8.internal.Bj.a(R8_3.3.28_2aaf796388b4e9f6bed752d926eca110512a53a3f09a8d755196089c1cfdf799:75)
1>	at com.android.tools.r8.internal.Bj.a(R8_3.3.28_2aaf796388b4e9f6bed752d926eca110512a53a3f09a8d755196089c1cfdf799:28)
1>	at com.android.tools.r8.internal.Bj.a(R8_3.3.28_2aaf796388b4e9f6bed752d926eca110512a53a3f09a8d755196089c1cfdf799:27)
1>	at com.android.tools.r8.internal.Bj.b(R8_3.3.28_2aaf796388b4e9f6bed752d926eca110512a53a3f09a8d755196089c1cfdf799:2)
1>	at com.android.tools.r8.D8.a(R8_3.3.28_2aaf796388b4e9f6bed752d926eca110512a53a3f09a8d755196089c1cfdf799:22)
1>	at com.android.tools.r8.D8.a(R8_3.3.28_2aaf796388b4e9f6bed752d926eca110512a53a3f09a8d755196089c1cfdf799:17)
1>	at com.android.tools.r8.internal.Bj.a(R8_3.3.28_2aaf796388b4e9f6bed752d926eca110512a53a3f09a8d755196089c1cfdf799:85)
1>	... 1 more
1>Caused by: com.android.tools.r8.internal.f: Type com.google.gson.FieldNamingPolicy$5 is defined multiple times: obj\Debug\net6.0-android33.0\lp\189\jl\libs\ED64959F88B22E6D.jar:com/google/gson/FieldNamingPolicy$5.class, obj\Debug\net6.0-android33.0\lp\194\jl\libs\2C5D90B3B25D7C1F.jar:com/google/gson/FieldNamingPolicy$5.class
1>	at com.android.tools.r8.internal.DT.a(R8_3.3.28_2aaf796388b4e9f6bed752d926eca110512a53a3f09a8d755196089c1cfdf799:14)
1>	at com.android.tools.r8.internal.DT.a(R8_3.3.28_2aaf796388b4e9f6bed752d926eca110512a53a3f09a8d755196089c1cfdf799:22)
1>	at com.android.tools.r8.internal.CN.a(R8_3.3.28_2aaf796388b4e9f6bed752d926eca110512a53a3f09a8d755196089c1cfdf799:33)
1>	at com.android.tools.r8.internal.CN.a(R8_3.3.28_2aaf796388b4e9f6bed752d926eca110512a53a3f09a8d755196089c1cfdf799:10)
1>	at java.base/java.util.concurrent.ConcurrentHashMap.merge(ConcurrentHashMap.java:2048)
1>	at com.android.tools.r8.internal.CN.a(R8_3.3.28_2aaf796388b4e9f6bed752d926eca110512a53a3f09a8d755196089c1cfdf799:6)
1>	at com.android.tools.r8.graph.B2$a.e(R8_3.3.28_2aaf796388b4e9f6bed752d926eca110512a53a3f09a8d755196089c1cfdf799:4)
1>	at com.android.tools.r8.dex.b.a(R8_3.3.28_2aaf796388b4e9f6bed752d926eca110512a53a3f09a8d755196089c1cfdf799:105)
1>	at com.android.tools.r8.dex.b.a(R8_3.3.28_2aaf796388b4e9f6bed752d926eca110512a53a3f09a8d755196089c1cfdf799:28)
1>	at com.android.tools.r8.D8.a(R8_3.3.28_2aaf796388b4e9f6bed752d926eca110512a53a3f09a8d755196089c1cfdf799:25)
1>	at com.android.tools.r8.D8.d(R8_3.3.28_2aaf796388b4e9f6bed752d926eca110512a53a3f09a8d755196089c1cfdf799:606)
1>	at com.android.tools.r8.D8.c(R8_3.3.28_2aaf796388b4e9f6bed752d926eca110512a53a3f09a8d755196089c1cfdf799:1)
1>	at com.android.tools.r8.internal.Bj.a(R8_3.3.28_2aaf796388b4e9f6bed752d926eca110512a53a3f09a8d755196089c1cfdf799:24)
1>	... 5 more
1>C:\Program Files\dotnet\packs\Microsoft.Android.Sdk.Windows\32.0.509\tools\Xamarin.Android.D8.targets(79,5): error : java.lang.RuntimeException: com.android.tools.r8.CompilationFailedException: Compilation failed to complete, origin: obj\Debug\net6.0-android33.0\lp\189\jl\libs\ED64959F88B22E6D.jar : com/google/gson/FieldNamingPolicy$5.class
1>Done building project "MauiApp9.csproj" -- FAILED.

**Comment by jpobst** [comment]
> @jpobst I think the original issue is fixed.

🎉
